### PR TITLE
Detect Group Policy on copy to let native copy go through

### PR DESCRIPTION
**Pull request by max-winderbaum**
set mode, reset selection after copy

### CODE REVIEW

File: src/component/handlers/edit/setClipboardData.js [hunk 1]
```diff
@@ -17,46 +17,59 @@ import type DraftEditor from 'DraftEditor.react';
...
-    var copyTrapRange = document.createRange();
-    copyTrapRange.selectNodeContents(editor._copyTrap);
-    selection.removeAllRanges();
-    selection.addRange(copyTrapRange);
+    const testClipboardString = 'draft js clipboard set works' + Math.floor(Math.random() * 100000);
+    clipboard.setData('Text', testClipboardString);
 
-    editor._copyTrap.focus();
+    if (clipboard.getData('Text') !== testClipboardString) {
+      console.log('bad stuff detected');
```
Comment (intentionally-left-nil) [inline src/component/handlers/edit/setClipboardData.js]: remove :-)

File: src/component/handlers/edit/setClipboardData.js [hunk 2]
```diff
@@ -17,46 +17,59 @@ import type DraftEditor from 'DraftEditor.react';
...
-    // through into that div
-    editor._copyTrap.innerHTML = html;
+  if (clipboard) {
 
-    var selection = window.getSelection();
-    var copyTrapRange = document.createRange();
-    copyTrapRange.selectNodeContents(editor._copyTrap);
-    selection.removeAllRanges();
-    selection.addRange(copyTrapRange);
+    const testClipboardString = 'draft js clipboard set works' + Math.floor(Math.random() * 100000);
```
Comment (intentionally-left-nil) [inline src/component/handlers/edit/setClipboardData.js]: no need to make a random number. Just pick a string and use it.

File: src/component/handlers/edit/setClipboardData.js [hunk 3]
```diff
@@ -17,46 +17,59 @@ import type DraftEditor from 'DraftEditor.react';
...
+      // We need to populate a "copy trap" div with the new HTML and switch focus so the native copy can go
+      // through into that div
+      editor.setMode('copy');
+      editor._copyTrap.innerHTML = html;
 
-    // Let the native event go through
-    return;
-  }
+      var selection = window.getSelection();
+      var originalRange = selection.getRangeAt(0).cloneRange();
```
Comment (intentionally-left-nil) [inline src/component/handlers/edit/setClipboardData.js]: This should check to see if there is a range in the first place. There should be... but you never know when it comes to browsers.

**Review by intentionally-left-nil — CHANGES_REQUESTED** [review]

File: src/component/handlers/edit/setClipboardData.js (hunk 2 above)
Comment (max-winderbaum) [reply src/component/handlers/edit/setClipboardData.js]: Yep. It's been quite a day.

**Review by intentionally-left-nil — APPROVED** [review]
ty

**Pull request merged**